sink/sink

```diff
@@ -116,6 +116,10 @@ def contains(expect, result, debug=True):
         return True
 
 
+def github_usr1_handler(sig, frame):
+    raise RuntimeError("Sink main thread received sigusr1 signal")
```

Comment:
Wording nag: you're dealing with processes, not threads  (please also change this in the commit message).

Also: why not create a custom exception type for this, or just sys.exit() on the spot?  sys.exit() is, itself, equivalent to just throwing a `SystemExit` exception.

Specifically, I'm worried that your `except` below might catch other `RuntimeError`s than the ones thrown from here.

Reply:
if I throw an exception straight from the handler it exits really weirdly. I could create a custom exception for this though.

Reply:
you *are* throwing an exception straight from the handler...

Reply:
to clarify this slightly: there's blanket try/excepts around the place which catch the `SystemExit` thrown from the handler, and either we have to catch `SystemExit` again and call `sys.exit` again from the `except`, or we have the custom exception.